dcpy/lifecycle/distribute/socrata.py

```diff
@@ -53,27 +62,31 @@ def dist_from_local(
             metadata=md,
             dataset_destination_id=dataset_destination_id,
             dataset_package_path=package_path,
-            publish=publish,
-            metadata_only=metadata_only,
+            publish=bool(publish),
+            metadata_only=bool(metadata_only),
         )
     except Exception as e:
         return f"Error pushing {md.attributes.display_name}, destination: {dest.id}: {str(e)}"
 
 
 def dist_from_local_all_socrata(
     package_path: Path,
-    **pub_kwargs,
+    **pub_kwargs: Unpack[PublishKwargs],
 ):
     """Distributes all Socrata destinations within a given metadata"""
     md = m.Metadata.from_path(package_path / "metadata.yml")
+    local_pub_kwargs = pub_kwargs.copy()
+    local_pub_kwargs.pop(
```

Comment:
nit, but is there a reason to not just do a standard if clause here? Since the value is not being set. Unless this is meant to be a variable and that's fixed in another commit

Reply:
Yeah, this was a little uncomfortable, due to the unpacking of the kwargs. Basically, I wanted to not modify the kwargs to never pass along the `metadata_path`, so what I would have wanted to do is, for immutability sake:
```
local_pub_kwargs = {k:v for k,v in kwagrs if k != "metadata_path"}
```
but `mypy` loses type of `local_pub_kwargs` in the above. (Could just slap a type annotation on it, but that feels like cheating). Def open to alternatives!

Reply:
Well i just meant

```python
if "metadata_path" in local_pub_kwargs:
    local_pub_kwargs.pop("metadata_path")
```

To get rid of the `else None`. But I think actually just
```python
local_pub_kwargs.pop("metadata_path", None)
```

Would work too